### DISCUSSION TITLE
Transform link url file extensions from .md to .html

### DISCUSSION THREAD

**Discussion by marvhen** [remarkjs/remark#667] [Q&A] [answered]
Surely there is an existing way to do this....what am I missing?  I can't find an appropriate plugin / option / setting to do this.

**Comment by ChristianMurphy** [comment, marked as answer]
This seems like an [XY Problem](https://meta.stackexchange.com/questions/66377/what-is-the-xy-problem), could you give content on what you are doing and your end goal? And include some information on which package(s) in remark/unified you are using?

**Reply by marvhen** [reply]
Indeed, I should have given more detail:
1. I have multiple markdown files in directory A which all have a .md extension.
2. The files reference each other via links such as `[Click here to go to File1](File1.md)` 
3. I execute `remark ./A/*.md -o ./B -r package.json` to transform the markdown files and output html files into directory B using the configuration I have provided in my package.json file.
4. The CLI runs and I get the html files all in the B directory with .html file extensions.
5. I expect the transformed files to have links such as `<a href="File1.html">Click here to go to File1</a>` but instead they are `<a href="File1.md">Click here to go to File1</a>`
6. The config in my package.json file is:

```
...
  "remarkConfig": {
    "plugins": [
      "html",
      "preset-lint-markdown-style-guide",
      ["lint-heading-style", "setext"],
      ["lint-maximum-line-length", false],
      ["lint-no-file-name-mixed-case", false],
      ["lint-list-item-indent", "space"],
      ["lint-list-item-spacing", false]
    ]
  },
...
```

**Reply by ChristianMurphy** [reply]
Gotcha, thanks for the context!
This is not something which `remark-html` does out of the box, two reasons for this:
1. it mainly focuses on converting markdown to html, not modifying the values of the content
2. not everything is guaranteed to be part of the remark pipeline and by extension be transformed

That said it is possible it can be done through a custom mdast to hast handler https://github.com/syntax-tree/mdast-util-to-hast#optionshandlers or as a remark or rehype plugin https://unifiedjs.com/learn

There's an existing plugin https://github.com/brechtcs/rehype-urls which may offer a good starting point for implementing the functionality
